lib/pbench/server/indexer.py

```diff
@@ -1180,42 +1180,59 @@ def _handle_iteration(self, iter_data, iter_name, iter_number, result_json):
         ):
             yield source, PbenchData.make_source_id(source), _parent, _type
 
-    # UID keyword pattern
+    # UID keyword pattern, non-greedy
     _uid_keyword_pat = re.compile(r"%\w*?%")
 
     @staticmethod
-    def expand_template(templ, d, run=None):
-        # match %...% patterns non-greedily.
-        s = templ
-        for m in re.findall(ResultData._uid_keyword_pat, s):
-            # m[1:-1] strips the initial and final % signs from the match.
+    def expand_uid_template(templ, d, run=None):
+        """Given a UID template string using %<keyword>% pattern identifiers,
+        lookup those keywords in the given dictionary and replacing those
+        keywords which have values.
+
+        There are two keywords which are treated specially:
+
+          * If the "benchmark_name" is not found in the given dictionary, a
+            second attempt to find a value using "name" will be attempted.
+
+          * If a additional `run` dictionary is provided, it'll be used on a
+            second attempt to find a "controller_host" value by using
+            "controller" in the `run` lookup.
+
+        The supported keyword values are one of: `str`, `int`, or `float`.  If
+        a keyword's value is not a supported type, the replacement is not
+        made.  For floats, 6 decimal places are used in the string conversion.
+
+        Returns a new string with as many keywords replaced as possible.
+
+        """
+        # Even though this is just a second name bound to the same string
+        # object, the `re.sub` method returns a new string object each time a
+        # change is made.
+        result = templ
+        for m in re.findall(ResultData._uid_keyword_pat, templ):
+            # Strips the initial and final % signs from the match.
             key = m[1:-1]
+            val = None
             try:
                 val = d[key]
             except KeyError:
-                if key == "benchmark_name":
-                    # Fix "benchmark_name" to try to find "name" in the
-                    # metadata if available (should be as we rename
-                    # benchmark_name to name).
-                    try:
+                try:
+                    if key == "benchmark_name":
+                        # Fix "benchmark_name" to try to find "name" in the
```

Comment:
You didn't change this, but 'Fix "benchmark_name"` isn't really the right phrasing since we're not "fixing" the key but rather trying an alternate key.